sdk/python/feast/pyspark/historical_feature_retrieval_job.py

```diff
@@ -1,19 +1,433 @@
+import abc
+import argparse
 import json
-from datetime import timedelta
-from typing import Any, Dict, List
+from datetime import datetime, timedelta
+from typing import Dict, List, NamedTuple, Optional
 
-from pyspark import SparkFiles
 from pyspark.sql import DataFrame, SparkSession, Window
 from pyspark.sql.functions import col, expr, monotonically_increasing_id, row_number
 
 
+class Source(abc.ABC):
+    """
+    Source for an entity or feature dataframe.
+
+    Attributes:
+        timestamp_column (str): Column representing the event timestamp.
+        created_timestamp_column (str): Column representing the creation timestamp. Required
+            only if the source corresponds to a feature table.
+        mapping (Optional[Dict[str, str]]): If present, the source column will be renamed
```

Comment:
should this be `field_map`?

Reply:
I am fine with making this `field_map`, but currently Oleksi's ingestion job is using `mapping` instead. Which one should i use then?

Reply:
Mapping seems a bit vague. The type is a map, so its already a mapping. @pyalex

Reply:
Renamed to `field_mapping` instead because that's what being used in the protobuf and the other part of the Feast SDK right now.